subitem/client.go

```diff
@@ -0,0 +1,192 @@
+	token := c.Key
+
+	if params != nil {
+		body = &stripe.RequestValues{}
+		body.Add("subscription", params.Sub)
+
+		if len(params.Plan) > 0 {
+			body.Add("plan", params.Plan)
+		}
+		if params.Quantity > 0 {
+			body.Add("quantity", strconv.FormatUint(params.Quantity, 10))
+		} else if params.QuantityZero {
+			body.Add("quantity", "0")
+		}
+
+		commonParams = &params.Params
+		params.AppendTo(body)
+	}
+
+	item := &stripe.SubItem{}
+	err := c.B.Call("POST", "/subscriptions_items", token, body, commonParams, item)
+	return item, err
+}
+
+// Get returns the details of a subscription.
+// For more details see https://stripe.com/docs/api#retrieve_subscription.
+func Get(id string, params *stripe.SubItemParams) (*stripe.SubItem, error) {
+	return getC().Get(id, params)
+}
+
+func (c Client) Get(id string, params *stripe.SubItemParams) (*stripe.SubItem, error) {
+	var body *stripe.RequestValues
+	var commonParams *stripe.Params
+
+	if params != nil {
+		body = &stripe.RequestValues{}
+		params.AppendTo(body)
+		commonParams = &params.Params
+	}
+
+	item := &stripe.SubItem{}
+	err := c.B.Call("GET", fmt.Sprintf("/subscriptions/%v", id), c.Key, body, commonParams, item)
+
+	return item, err
+}
+
+// Update updates a subscription's properties.
+// For more details see https://stripe.com/docs/api#update_subscription.
+func Update(id string, params *stripe.SubItemParams) (*stripe.SubItem, error) {
+	return getC().Update(id, params)
+}
+
+func (c Client) Update(id string, params *stripe.SubItemParams) (*stripe.SubItem, error) {
+	var body *stripe.RequestValues
+	var commonParams *stripe.Params
+	token := c.Key
+
+	if params != nil {
+		body = &stripe.RequestValues{}
+
+		if len(params.Plan) > 0 {
+			body.Add("plan", params.Plan)
+		}
+
+		if params.NoProrate {
+			body.Add("prorate", strconv.FormatBool(false))
+		}
+
+		if params.Quantity > 0 {
+			body.Add("quantity", strconv.FormatUint(params.Quantity, 10))
+		} else if params.QuantityZero {
+			body.Add("quantity", "0")
+		}
+
+		if params.ProrationDate > 0 {
+			body.Add("proration_date", strconv.FormatInt(params.ProrationDate, 10))
+		}
+
+		commonParams = &params.Params
+		params.AppendTo(body)
+	}
+
+	sub := &stripe.Sub{}
+	err := c.B.Call("POST", fmt.Sprintf("/subscriptions_items/%v", id), token, body, commonParams, sub)
+
+	return sub, err
+}
+
+// Cancel removes a subscription.
+// For more details see https://stripe.com/docs/api#cancel_subscription.
+func Del(id string, params *stripe.SubItemParams) (*stripe.SubItem, error) {
+	return getC().Cancel(id, params)
+}
+
+func (c Client) Del(id string, params *stripe.SubItemParams) (*stripe.SubItem, error) {
+	var body *stripe.RequestValues
+	var commonParams *stripe.Params
+
+	if params != nil {
+		body = &stripe.RequestValues{}
+		params.AppendTo(body)
+		commonParams = &params.Params
+	}
+
+	item := &stripe.Sub{}
+	err := c.B.Call("DELETE", fmt.Sprintf("/subscriptions_item/%v", id), c.Key, body, commonParams, item)
```

Comment:
And same with here (and the other CRUD endpoints). The patch merged to Ruby was for `subscription_items` (i.e. non-plural subscription), so one of these must be wrong.